src/Service.Tests/Authorization/GraphQL/CreateMutationAuthorizationTests.cs

```diff
@@ -0,0 +1,390 @@
+                                          }";
+
+            // The 'test_role_with_excluded_fields_on_create' role does not have create permissions on
+            // stocks.piecesAvailable field and hence the authorization check should fail.
+            await ValidateRequestIsAuthorized(
+                graphQLMutationName: createOneStockMutationName,
+                graphQLMutation: nestedCreateOneStockWithoutPiecesAvailable,
+                isAuthenticated: true,
+                clientRoleHeader: "test_role_with_excluded_fields_on_create",
+                expectedResult: "");
+        }
+
+        /// <summary>
+        /// Test to validate that a 'create multiple' mutation request can only execute successfully when the user, in addition to having
+        /// create permission for all the entities involved in the create mutation, has the create permission for all the columns
+        /// present for each entity in the mutation.
+        /// If the user does not have any create permission on one or more column belonging to any of the entity in the
+        /// multiple-create mutation, the request will fail during authorization check.
+        /// </summary>
+        [TestMethod]
+        public async Task ValidateAuthZCheckOnColumnsForCreateMultipleMutations()
+        {
+            string createMultipleStockMutationName = "createStocks";
+            string createMultipleStocksWithPiecesAvailable = @"mutation {
+                                            createStocks(
+                                                items: [
+                                                  {
+                                                    categoryid: 1,
+                                                    pieceid: 2,
+                                                    categoryName: ""xyz""
+                                                    piecesAvailable: 0,
+                                                    stocks_price:
+                                                      {
+                                                        is_wholesale_price: true
+                                                        instant: ""1996-01-24""
+                                                      }
+                                                  }
+                                             ])
+                                             {
+                                                items
+                                                {
+                                                   categoryid
+                                                   pieceid
+                                                }
+                                             }
+                                          }";
+
+            // The 'test_role_with_excluded_fields_on_create' role does not have create permissions on
+            // stocks.piecesAvailable field and hence the authorization check should fail.
```

Comment:
Seems that all the tests in this file check for permissions on either:
- top level entity
- top level entity fields
- sub-entity

I don't see tests that cover permissions check on 
- sub-entity fields. (which would validate behavior of recursive authZ calls when processing the relationship `stocks_price` , or `publishers` in other tests)

Reply:
added